### PR TITLE
fix: do not print stack trace for invalid option

### LINKED ISSUE #2891: Stack trace is printed when providing invalid option choice
```
$ prettier --version
1.7.0

$ echo 'test' | prettier --parser foo
stdin: Error: Couldn't resolve parser "foo"
    at resolveParseFunction (/home/lydell/bugs/prettier/node_modules/prettier/index.js:3774:13)
    at Object.parse (/home/lydell/bugs/prettier/node_modules/prettier/index.js:3782:25)
    at formatWithCursor (/home/lydell/bugs/prettier/node_modules/prettier/index.js:21730:22)
    at Object.formatWithCursor (/home/lydell/bugs/prettier/node_modules/prettier/index.js:21991:12)
    at format (/home/lydell/bugs/prettier/node_modules/prettier/bin/prettier.js:16346:21)
    at getStream.then.input (/home/lydell/bugs/prettier/node_modules/prettier/bin/prettier.js:16411:19)
    at <anonymous>
    at process._tickCallback (internal/process/next_tick.js:188:7)
```

### PR DESCRIPTION
**Pull request by ikatyang**
Fixes #2891

before/after

image

### CODE REVIEW

File: src/parser.js [hunk 1]
```diff
@@ -38,7 +38,7 @@ function resolveParseFunction(opts) {
       return eval("require")(path.resolve(process.cwd(), opts.parser));
     } catch (err) {
       /* istanbul ignore next */
-      throw new Error(`Couldn't resolve parser "${opts.parser}"`);
+      throw `Couldn't resolve parser "${opts.parser}"`;
```
Comment (lydell) [inline src/parser.js]: This works, but is it the correct fix? Do we want to catch errors somewhere instead?

**Comment by lydell** [comment]
Nice!

There are two occurrences of throwing strings in cli-util.js as well, if you feel like looking if we can use a custom error there as well. :)

```
src/cli-util.js
117:      throw "prettier(input) !== prettier(prettier(input))\n" + diff(pp, pppp);
128:        throw "ast(input) !== ast(prettier(input))\n" +
```

File: src/errors.js [hunk 2]
```diff
@@ -0,0 +1,5 @@
+"use strict";
+
+class ConfigError extends Error {}
```
Comment (azz) [inline src/errors.js]: I seem to recall there being issues in some engines with subclassing builtins but I can't remember if it affects `Error`...

Comment (lydell) [reply src/errors.js]: Are you perhaps thinking about that it used to be more or less impossible to subclass builtins before ES6 (and that it therefore can’t be transpiled correctly)?

Comment (azz) [reply src/errors.js]: Ah, yeah that could be it. If someone was to downlevel compile this to es5 for browser use will it still work?

I've always done it this way. Avoids nasty `instanceof` problems as well.

```js
function createConfigError(message) {
  const error = new Error(message);
  error.type = "ConfigError";
  return error;
}

function isConfigError(error) {
  return error && error.type === "ConfigError";
}
```

Comment (lydell) [reply src/errors.js]: Extending `Error` seems to work flawlessly in Node.js 4+. Let’s stick with that until somebody brings up an issue.

**Pull request merged**